splash/utils.py

```diff
@@ -197,3 +200,35 @@ def ensure_tuple(val):
     if not isinstance(val, tuple):
         return (val,)
     return val
+
+
+def get_id():
+    return str(uuid1())
+
+
+def traverse_obj(obj, predicate, convert, max_depth=100):
+    if max_depth <= 0:
+        raise ValueError("Can't traverse through object: depth limit is reached")
+
+    if obj is None:
+        return None
+
+    if predicate(obj):
+        return convert(obj)
+
+    if isinstance(obj, dict):
+        return {
+            key: traverse_obj(value, predicate, convert, max_depth=max_depth - 1)
```

Comment:
This function is un utils.py, its description says nothing about HTML elements. If it is only for traversing specific input (a Python data structure which possibly contain HTML Element wrappers?) then it could be better to move it somewhere else.